src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -253,16 +253,29 @@ impl TargetInfo {
                 res
             };
 
-            let cfg = lines
-                .map(|line| Ok(Cfg::from_str(line)?))
-                .filter(TargetInfo::not_user_specific_cfg)
-                .collect::<CargoResult<Vec<_>>>()
-                .with_context(|| {
-                    format!(
-                        "failed to parse the cfg from `rustc --print=cfg`, got:\n{}",
-                        output
-                    )
-                })?;
+            let cfg = {
+                let mut res = Vec::new();
+                loop {
+                    match lines.next() {
+                        Some(line) => {
+                            let cfg = Cfg::from_str(line).with_context(|| {
+                                format!(
+                                    "failed to parse the cfg from `rustc --print=cfg`, got:\n{}",
+                                    output
+                                )
+                            })?;
+                            if TargetInfo::not_user_specific_cfg(&cfg) {
+                                res.push(cfg);
+                            }
+                        }
+                        None if res.is_empty() => {
+                            return error_missing_print_output("cfgs", &process, &output, &error)
```

Comment:
I'm not seeing this in the old code.  What is this replacing?

Reply:
I following what was done above for the `--print=split-debuginfo`, I think it's a safe guard just in case there is no cfgs. I can remove it if desired.

Reply:
A behavior change like that should probably be called out.

Is there ever a reason rustc will print nothing?

Reply:
Revert to not error if no cfgs printed.